cli/src/formatter.rs

```diff
@@ -132,7 +132,17 @@ where
             if let Some(heading) = self.heading.take() {
                 write!(formatter.labeled("heading"), "{heading}")?;
             }
-            formatter.write_fmt(args)
+            let msg = args.to_string();
+            if msg.matches('\n').nth(1).is_some() {
+                // message is multi-line, print it in an indented block
+                writeln!(formatter)?;
+                for line in msg.lines() {
+                    writeln!(formatter, "  {line}")?;
+                }
+                Ok(())
+            } else {
```

Comment:
Since `.write_fmt()` may be called more than once, we'll need to keep track of whether `'\n'` was printed or not. It would be easier to do indentation by caller.

(fwiw, I don't know if `writeln!()` macro is guaranteed to be expanded to a single `.write_fmt()` call, but apparently the current implementation is.)

Reply:
Hm. If `write_fmt` is called multiple times for a single log message, and we have to track whether `\n` was already printed or not, then inserting a first line break is not possible.

So this format would _not_ work:
```
Warning:
  foo
  bar
```

but this one would:
```
Warn: foo
      bar
```

If it's better to not do this centrally, we can drop this PR. I can still go through the log messages manually and break some of the long ones onto multiple lines, using the `indoc` macros to ensure the indentation is appropriate.

Reply:
Perhaps, the caller can opt in to the multi-line format by `.multiline()` or `.indented()`? I don't know if that's worth the effort, though.

BTW, indented text would look horrible if it's wrapped by terminal. We'll need to keep lines short.

Reply:
For me, many commands use the pager if the lines are too long. But not all of them, and it could be environment dependent, not sure. I agree we need to make the lines short if we're going to add the indentation.

My plan is to go through a bunch of the really long single-line log messages and hard-wrap them to 80 characters or so. I think that would be an improvement for the majority of users.